packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -597,6 +596,19 @@ export const Table = <
     visibleColumns,
     setGlobalFilter,
   } = instance;
+
+  let headerGroups = _headerGroups;
+
+  if (columns.length === 1 && 'columns' in columns[0]) {
```

Comment:
Is there a way to test if this is added only in dev environment?

Reply:
1. add an example in vite playground that uses top-level HeaDer
2. when you run `yarn dev` and open `localhost:1701`, you should see the log.
3. stop the dev servers and run `yarn dev --filter=vite-playground`
4. now run `yarn workspace vite-playground preview`, then open `localhost:4173`

Reply:
I meant like unit tests for that. But I guess it's not that bad if it's not showing for some reason (if we break some code in the future or something).

Reply:
Storybook also shows this warning.